src/components/List.js

```diff
@@ -0,0 +1,162 @@
+
+const ToDoBlock = styled.div`
+  background-color: white;
+  border-radius: 10px;
+  width: 100%;
+  height: 50vh;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  overflow-y: auto;
+  overflow-x: hidden;
+
+  /* 웹킷 기반 브라우저를 위한 스타일 */
+  &::-webkit-scrollbar {
+    width: 5px;
+  }
+
+  /* 파이어폭스 브라우저를 위한 스타일 */
+  scrollbar-width: thin;
+
+  @media (max-width: 768px) {
+    width: 100%;
+  }
+`;
+
+const ItemContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 10px 10px 10px 30px;
+  position: relative;
+  min-height: 40px; /* 버튼 컨테이너 상관없이 일관된 높이 */
+
+  &::after {
+    content: ''; /* 가상 요소에 내용이 없음을 명시 */
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    width: calc(100% - 30px); /* 전체 너비에서 30px만큼 줄임 */
+    margin-left: 30px;
+    border-bottom: 1px solid #ccc;
+  }
+
+  css @media (max-width: 768px) {
+    padding: 5px; 5px; 5px; 15px;
+    &::after {
+      width: calc(100% - 20px);
+      margin-left: 20px;
+    }
+  }
+`;
+
+const TextContainer = styled.div`
+  width: 100%;
+  font-size: 1em;
+  text-decoration: ${(props) => (props.completed ? 'line-through' : 'none')};
+  word-wrap: break-word;
+  white-space: normal;
+  cursor: pointer;
+
+  @media (max-width: 768px) {
+    font-size: 0.9em;
+  }
+`;
+
+// 애니메이션 효과
+const SlideIn = keyframes`
+  from {
+    transform: translateX(100%);
+  }
+
+  to {
+    transform: translateX(0);
+  }
+`;
+
+const ButtonsContainer = styled.div`
+  display: flex;
+  animation: ${SlideIn} 0.5s forwards; // 0.5초 동안 애니메이션 실행
+`;
+
+const DeleteButtonImg = styled.img`
+  cursor: pointer;
+  width: 20px;
+  height: 20px;
+`;
+
+const CheckButtonImg = styled.img`
+  width: 20px;
+  height: 20px;
+  cursor: pointer;
+  margin-right: 7px;
+`;
+
+// React.memo를 사용하여 컴포넌트의 불필요한 렌더링을 방지
+export default React.memo(function List({ toDoData, setToDoData }) {
+  const [showButtons, setShowButtons] = useState(false);
+  const [selectedTextId, setSelectedTextId] = useState(null);
+
+  // 텍스트 컨테이너 클릭 핸들러
+  const handleTextClick = (id) => {
+    if (id === selectedTextId) {
+      // 같은 TextContainer 클릭 시 showButton 상태 변경
+      setShowButtons(!showButtons);
+    } else {
+      // 다른 TextContainer가 클릭되면, showButtons를 true로 설정하고 selectedTextId를 업데이트
+      setShowButtons(true);
+      setSelectedTextId(id);
+    }
+  };
+
+  // 할 일 목록 삭제하는 함수
+  const handleDelete = (id) => {
+    let newToDoData = toDoData.filter((item) => item.id !== id);
+    setToDoData(newToDoData);
+  };
+
+  // 할 일 목록의 완료 상태 변경하는 함수
+  const handleComplete = (id) => {
+    let newToDoData = toDoData.map((item) => {
+      if (item.id === id) {
+        return { ...item, completed: !item.completed };
+      }
+      return item;
+    });
+    setToDoData(newToDoData);
+  };
+
+  return (
+    <ToDoBlock>
+      {toDoData.map((item) => (
+        <ItemContainer key={item.id}>
+          <TextContainer
+            completed={item.completed}
```

Comment:
styled component에 props 전달까지 최곤데요 ? ㅎㅎ

그런데 한가지 주의해야할 점이 있는데요 바로 **표준 속성 워닝**입니당 ㅎㅎ

> 표준 속성 워닝이란?
> React와 styled-components에서는 표준 HTML 속성이 아닌 사용자 정의 속성이 DOM 엘리먼트에 전달되는 것을 기본적으로 허용하지 않아서 발생하는 에러 

따라서 styled component 5.1버전 이상부터는 $를 prefix로 써서 warning을 없애고 DOM에 불필요한 속성이 전달되는 것을 막아주었어요 !! 

혹시나 구체적인 내용 더 확실히 알고 싶을 수 있으니 [관련 아티클 ](https://velog.io/@jiwonyyy/Styled-components에서-props-똑똑하게-넘겨주기-feat.-Received-true-for-a-non-boolean-attribute#왜-warning이-발생했을까) 첨부할게요 > <

Reply:
이 부분은 아래처럼 바꿔주면 되겠죠 ? 😆
```suggestion
          <TextContainer
            $completed={item.completed}
```

Reply:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ @Rose-my 저 민영님한테 코드리뷰 받고싶어요.... 넘 정성스러움 민영님이 알려주신 부분 아마 이미 브라우저 콘솔창에서 warning으로 뜨고 있을 거예요! 개발하실 때 콘솔창 확인 자주 해주시면 많은 걸 놓치지 않을 수 있답니다 :)

Reply:
헉 덕분에 남겨주신 블로그 잘 읽어봤어요!! 또 하나 배워갑니다.!!